File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        return daily(nums, x);
    }
private:
    int daily(vector<int>& nums, int x) {
        int sum = 0, i = 0, j = 0;
        int n = nums.size();
        int ans = INT_MAX;
        
        sum = accumulate(nums.begin(), nums.end(), 0);
        while(i <= j) {
            if(sum >= x) {
                if(sum == x)
                    ans = min(ans, n + i - j);
                
                if(j < n)
                    sum -= nums[j++];
                else
                    break;
            }
            else
                sum += nums[i++];
        }
        if(ans == INT_MAX)
            return -1;
        return ans;
           
    }
};

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        _sum = sum(nums)
        i, j = 0, 0
        n, ans = len(nums), float('inf')
        
        while i <= j:
            if _sum >= x:
                if _sum == x:
                    ans = min(ans, n + i - j)
                
                if j < n:
                    _sum -= nums[j]
                    j += 1
                else:
                    break
            else:
                _sum += nums[i]
                i += 1
        
        return ans if ans != float('inf') else -1
                    